data/blog/2022-04-08.robur.md

```diff
@@ -0,0 +1,47 @@
+
+The code for [CalDAV](https://github.com/roburio/caldav) is also released to opam-repository, the code for [CardDAV](https://git.robur.io/linse/carddav) is not yet integrated nor released.
+
+### DNS Projects
+
+Robur engineers have created robust DNS Projects, like our [‘Let’s Encrypt’](https://letsencrypt.org/)-Certified [DNS Solver](https://builds.robur.coop/job/dns-letsencrypt/), a [DNS Resolver](https://builds.robur.coop/job/dns-resolver/), and an authoritative [DNS Server](https://builds.robur.coop/job/dns-primary-git/). As a refresher, users navigate the Internet using domain names, addresses in cyberspace; DNS stands for Domain Name System. These systems take the domain name (something easy to remember, like robur.coop), and reroute them to the IP address (not easily remembered, like 193.30.40.138). The term *IP address* is short for Internet Protocol address, and its numbers point to a specific server, city, state/region, postal code, etc. In other words, it’s the actual machine’s address. These IP addresses take users to the website's files, and they’re what enable you to send and receive emails, too. DNS stores all these key values (with caching) in a decentralised hierarchy, making it fault-tolerant to minimise problems.
+
+Robur’s authoritative DNS server delegates responsibility for a specific domain and provides mapping information for it, ensuring that the user gets to the correct IP address. At the other end of the process, Robur’s DNS resolver finds the exact server to handle the user’s request. To keep the codebase minimal for security and simplicity, they included only the elements absolutely necessary. The let's encrypt unikernel waits for certificate requests in the zones (encoded as TLSA records), and uses let's encrypt DNS challenge to have these signed to certificates, again stored in the zones. Of course, certificates that will soon become invalid will be updated by dns-letsencrypt-secondary. This unikernel does not need any persistent storage, it leaves this to the dns-primary-git.
+
+We’ve been developing these DNS projects since 2017, which serve a multitude of functions in the Robur ecosystem. Our domains (like nqsb.io and robur.coop) use our DNS server as an authoritative server, and we also use a caching resolver for our biannual [MirageOS retreats](http://retreat.mirage.io) in Marrakech. Additionally, any MirageOS unikernel can use our client to resolve domain names - using the dns-client devices, or [happy-eyeballs](https://github.com/roburio/happy-eyeballs), so it’s also beneficial to the OCaml community at large. Robur uses expressive OCaml types (GADT), so we can ensure a given query has a specific shape, like an address record query results in a set of IPv4 addresses.
```

Comment:
```suggestion
We’ve been developing these DNS projects since 2017, and they serve a multitude of functions in the Robur ecosystem. Our domains (like nqsb.io and robur.coop) use our DNS server as an authoritative server, and we also use a caching resolver for our biannual [MirageOS retreats](http://retreat.mirage.io) in Marrakech. Additionally, any MirageOS unikernel can use our client to resolve domain names - using the dns-client devices, or [`happy-eyeballs`](https://github.com/roburio/happy-eyeballs), so it’s also beneficial to the OCaml community at large. Robur uses expressive OCaml types (GADT), so we can ensure a given query has a specific shape, like an address record query results in a set of IPv4 addresses.
```